README.md

```diff
@@ -1,4 +1,217 @@
+
+```bash
+conda create -y --name marabou python=3.8
+conda activate marabou
+cd ~/marabou/vnncomp
+bash install_tool.sh
+cp -r ~/marabou/opt ~/
+```
+
+---
+
+### Step 2: Run Verification
+
+#### 2.1: Run alpha-beta-CROWN
+
+To run **alpha-beta-CROWN** with activation split:
+
+```bash
+python run_all_verification.py --verifier abcrown --model_folder <path_to_soundness_bench>
+```
+
+To run **alpha-beta-CROWN** with input split:
+
+```bash
+python run_all_verification.py --verifier abcrown --model_folder <path_to_soundness_bench> --split_type input
+```
+
+#### 2.2: Run NeuralSAT
+
+To run **NeuralSAT** with activation split:
+
+```bash
+python run_all_verification.py --verifier neuralsat --model_folder <path_to_soundness_bench>
+```
+
+To run **NeuralSAT** with input split:
+
+```bash
+python run_all_verification.py --verifier neuralsat --model_folder <path_to_soundness_bench> --split_type input
+```
+
+#### 2.3: Run PyRAT
+
+To run **PyRAT**:
+
+```bash
+python run_all_verification.py --verifier pyrat --model_folder <path_to_soundness_bench>
+```
+
+#### 2.4: Run Marabou
+
+We use **Docker** to run **Marabou**. Please specify the container name in the script. If you want to run a specific version of **Marabou**, just specify the name of the container corresponding to its version.
+
+```bash
+# Replace <marabou_container> with your Marabou container.
+python run_all_verification.py --verifier marabou --model_folder <path_to_soundness_bench> --container_name <marabou_container>
+```
+
+### Step 3: Count results in results.csv
+The `run_all_verification.py` script generates `<toolname>_results.csv` files in the `./results` directory. You can use the `eval.py` script to count the results in these files.
+
+```bash
+python eval.py <path to results.csv> <path to output.csv>
+```
+
+## Train New Models
+We provide an easy pipeline for users to train new models that contain hidden adversarial examples.
+
+### File Usage Overview
+* `synthetic_data_generation.py` generates synthetic data with pre-defined hidden counterexamples
+* `adv_training.py` trains models on the synthetically generated dataset using a two-objective training framework
+* `cross_attack_evaluation.py` uses very strong adversarial attacks to make sure that no trivial counterexamples can be found and saves those instances into VNNLIB files
+* `models` contain the definitions of all NN architectures we use in SoundnessBench
+<!-- * `run_all_verification.py` provides an easy way to run verification on SoundnessBench for four verifiers: alpha-beta-CROWN, Marabou, NerualSAT, and PyRat, results will be saved in `results` folder. -->
+
+### Install Dependencies
+TODO
+
+### Detailed Walkthrough
+To train a new model, you can run the following command, which will first call `synthetic_data_generation.py` to generate the synthetic dataset used for training, and then apply a two-objective training framework (see our paper for details) to the model.
+```bash
+python adv_training.py
```

Comment:
To make things easier for users, I think we don't need to include options which normally don't have to be changed, such as `--attack`, etc.